src/track/beatmap.cpp

```diff
@@ -476,10 +494,8 @@ std::unique_ptr<BeatIterator> BeatMap::findBeats(double startSample, double stop
         return std::unique_ptr<BeatIterator>();
     }
 
-    Beat startBeat, stopBeat;
-    startBeat.set_frame_position(
-            static_cast<google::protobuf::int32>(samplesToFrames(startSample)));
-    stopBeat.set_frame_position(static_cast<google::protobuf::int32>(samplesToFrames(stopSample)));
+    Beat startBeat = beatFromFramePos(mixxx::audio::FramePos::fromEngineSamplePos(startSample));
```

Comment:
beatFromFramePos(mixxx::audio::FramePos::fromEngineSamplePos(value)) is invoked multiple times. Extract as a small, inlined utility function?

I noticed that there is one special case where `.toFullFrames()` is invoked before passing the value into `beatFromFramePos()`. It might make sense to extract this as a separate function with a dedicated name to isolate the occurrences and have all possible variations close to each other.

Reply:
beatFromFramePos always strips away fractional frame positions. In fact, we can always invoke `.toFullFrames()`. But I wasn't sure if that is intended.

The location where "toFullFrames" is called explicitly is the location where we previously called "std::floor" explicitly.
I wasn't sure if that is just coincidence or if it's indeed desired. But if I uncomment the debug assertion that checks that the frame position is not fractional, some tests fail. The qeustion is: Do we always want to discard frame positions silently, or should be make it explicit and re-add the assertion?

But that kind of investigation is out of scope for this PR IMHO.

Reply:
The name of the function is also not very intuitive. But let's just leave it as is for the moment.

If there are known issues that must be resolved before a release we might also use FIXME as a marker to distinguish *mandatory* from *optional* TODOs, e.g. to either enable the debug assertion or update the documentation. This could be handy to quickly find these spots when we decide to merge unfinished work.